packages/plugin-ext/src/plugin/file-system.ts

```diff
@@ -99,4 +117,26 @@ export class FileSystemExtImpl implements FileSystemExt {
         const opts = { create: true, overwrite: true };
         return Promise.resolve(this.fsProviders.get(handle)!.writeFile(uri, buffer, opts));
     }
+
+    $delete(handle: number, resource: UriComponents, options: { recursive: boolean }): Promise<void> {
+        this.checkProviderExists(handle);
+        const uri = URI.revive(resource);
+        return Promise.resolve(this.fsProviders.get(handle)!.delete(uri, options));
+    }
+
+    $rename(handle: number, source: UriComponents, target: UriComponents, options: { overwrite: boolean }): Promise<void> {
+        this.checkProviderExists(handle);
+        const sourceUri = URI.revive(source);
+        const targetUri = URI.revive(target);
+        return Promise.resolve(this.fsProviders.get(handle)!.rename(sourceUri, targetUri, options));
```

Comment:
[quality]: Let's try to have fewer `!`. Thoughts?

Reply:
refactored so now there are no `!`